File: user-app/src/main/java/com/github/dkurata38/liblio/user_app/HomeController.java
package com.github.dkurata38.liblio.user_app;

import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import com.github.dkurata38.liblio.user_app.security.LiblioUser;

@Controller
public class HomeController {
    @GetMapping
    public String index(
        @AuthenticationPrincipal LiblioUser liblioUser
    ) {
        if (liblioUser == null) {
            return "index_unauthenticated";
        }
        return "index";
    }
}

File: user-app/src/main/kotlin/com/github/dkurata38/liblio/user_app/HomeController.kt
package com.github.dkurata38.liblio.user_app

import com.github.dkurata38.liblio.user_app.security.LiblioUser
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping

@Controller
class HomeController {
    @GetMapping
    fun index(@AuthenticationPrincipal liblioUser: LiblioUser?): String {
        return if (liblioUser == null) {
            "index_unauthenticated"
        } else "index"
    }
}
